### DISCUSSION TITLE
Non-commercialisation and sharing Bounties?

### DISCUSSION THREAD

**Discussion by PeerRich** [NostrGit/NostrGit#72] [General]
- I dont think NostrGit should ever be incorporated/funded by VC
- However, I think we should accept bounties and fully distribute them with contributors (Jack has a 10 BTC bounty for a nostr-based Git)

Questions:
- Any ideas what a good governance form other than a corporation this could turn int?
- how (or who) to decide how much of the bounty goes to what person?
- should we start tracking PRs with a point system (1 point = simple PR, 3 points medium PR, 5 points hard PR) and distribute donations/bounties accordingly?

**Comment by PeerRich** [comment]
once discussed, this should move into the vision.md

**Comment by brycedev** [comment]
great question. i wish this (or similar) was functional: https://docs.allo.gitcoin.co/getting-started/introduction

**Comment by cypherhoodlum** [comment]
> I dont think NostrGit should ever be incorporated/funded by VC

Agree completely.

> Any ideas what a good governance form other than a corporation this could turn int?

It's very early to discuss this, but I feel the NostrGit project should always stay as open as possible, meaning anyone can contribute and fork the code, collect zaps as individual developers, and use the code however they see fit within the license. That being said, there will be those who participate more actively as developers and maintainers of the codebase, and those people should in my opinion distribute the sats zapped to the NostrGit project itself between themselves. That hypothetical project funding can also be used to zap casual contributors from within the project, if need be.

> how (or who) to decide how much of the bounty goes to what person?
should we start tracking PRs with a point system

I think some kind of a point system would be fair, keeping in mind that many people can work on a PR, and probably not only the person submitting it has contributed to it. Another thing is that a lot of work goes into maintaining a project itself, apart from the code. And then the question turns into: who gets to decide how many points a given feature/PR is worth? And how would it be tracked in a trustless manner? A difficult problem for sure.

**Comment by cypherhoodlum** [comment]
Maybe the distribution of the hypothetical bounty should be decided with some kind of a voting mechanism where everyone who has actively participated in the project gives each other a score and they get averaged out, giving a percentage for each? Is that stupid? I don't think we'll whip out a completely fair and trustless mechanism for this anytime soon.

**Comment by PeerRich** [comment]
i think the zaps functionality will solve a lot of these issues

**Comment by cypherhoodlum** [comment]
Let's hope so.

**Comment by PeerRich** [comment]
> I think some kind of a point system would be fair, keeping in mind that many people can work on a PR, and probably not only the person submitting it has contributed to it. Another thing is that a lot of work goes into maintaining a project itself, apart from the code. And then the question turns into: who gets to decide how many points a given feature/PR is worth? And how would it be tracked in a trustless manner? A difficult problem for sure.

yeah pretty hard thing to solve.

just in case we win the jack bounty, i dont mind "collecting" incoming bounties initially and only redistributing to contributors

my main job is my priority and i'm not looking to make a quick buck here -- I really want a nostr-based git

**Comment by PeerRich** [comment]
> some kind of a voting mechanism

i think thats the best approach

**Comment by PeerRich** [comment]
> just in case we win the jack bounty, i dont mind "collecting" incoming bounties initially and only redistributing to contributors


or a nostrgit multisig

**Comment by cypherhoodlum** [comment]
> just in case we win the jack bounty, i dont mind "collecting" incoming bounties initially and only redistributing to contributors

I think that is how it will go in practise if (when) we get that far.

**Comment by cypherhoodlum** [comment]
We [decided](https://github.com/NostrGit/NostrGit/discussions/95) to not use Discussions anymore. Let's continue the discussion at #122  :heart:
